### PR TITLE
feat:user profile

### PR DESCRIPTION
**Pull request by tushargoyal812**
https://user-images.githubusercontent.com/69681896/168491428-597df1ec-50c3-47d2-a9e4-c582836a2289.mp4

### CODE REVIEW

File: my-tweet/src/context/post-context.js [hunk 1]
```diff
@@ -11,7 +11,9 @@ const PostProvider=({children})=>{
     const [allUsers,setAllUsers]=useState()
     const [userFollow,setUserFollow]=useState()
     const [userUnfollow,setUserUnfollow]=useState({})
-    return(<PostContext.Provider value={{posts,setPosts,newPostData,setNewPostData,login,setLogin,userBookmarks,setUserBookmarks,allUsers,setAllUsers,userFollow,setUserFollow,userUnfollow,setUserUnfollow}}>{children}</PostContext.Provider>)
+    const [loggedInUser,setLoggedInUser]=useState()
+    const [editUserData,setEditUserData]=useState({firstName:"",lastName:"",username:"",bio:"",portfolioURL:""})
+    return(<PostContext.Provider value={{posts,setPosts,newPostData,setNewPostData,login,setLogin,userBookmarks,setUserBookmarks,allUsers,setAllUsers,userFollow,setUserFollow,userUnfollow,setUserUnfollow,loggedInUser,setLoggedInUser,editUserData,setEditUserData}}>{children}</PostContext.Provider>)
```
Comment (bajajcodes) [inline my-tweet/src/context/post-context.js:16]: These set of states can be clubbed into single state as object, if suits your project requirements and constraints,

File: my-tweet/src/components/userProfile.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { EditProfileModal } from "./edit-profile-modal"
+
+export const UserProfileComponent=()=>{
+    const { isOpen, onOpen, onClose } = useDisclosure()
+    const {loggedInUser,setLoggedInUser,userFollow}=usePost()
+
+    const getUserData=()=>{
+        const userData=JSON.parse(localStorage.getItem("userName"))
+        setLoggedInUser(userData)
+        console.log(userData,"userData");
```
Comment (bajajcodes) [inline my-tweet/src/components/userProfile.js:14]: This console statement is to be removed.

File: my-tweet/src/pages/following.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        <Flex>
+        <Sidebar/>
+        <Box border='1px' w='38rem'>
+            {userFollow&&userFollow.following.map(data=>
+                <Flex m='2rem' align='center' justify='space-between' key={data._id}>
+                    <Box>
+                    <Heading size='md'>{`${data.firstName} ${data.lastName}`}</Heading>
+                    <Box>@{data.username}</Box>
+                    </Box>
+                    {userFollow.following.some(item=>item._id===data._id)?<Button onClick={()=>unfollowHandler(data._id,setUserFollow)} size='xs' colorScheme='teal' variant='outline'>Following</Button>:<Button onClick={()=>followHandler(data._id,setUserFollow)} colorScheme='blue' size='xs'>Follow</Button>}
```
Comment (bajajcodes) [inline my-tweet/src/pages/following.js:24]: Use code formatter to make it look more pleasent.

**Review by bajajcodes — APPROVED** [review]
- Good use of abstractions and code is self descriptive as well.
- Left few comments and code is great to merge :)

File: my-tweet/src/components/edit-profile-modal.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        const token=localStorage.getItem("user")
+        try {
+            const response=await axios.post(`/api/users/edit`,{
+                userData:editUserData
+            },{
+                headers:{
+                  authorization:token
+                }
+              })
+              console.log(response,"response");
```
Comment (deepak29-git) [inline my-tweet/src/components/edit-profile-modal.js:33]: remove console

File: my-tweet/src/components/edit-profile-modal.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  } from '@chakra-ui/react'
+import axios from "axios"
+import { usePost } from "../context/post-context"
+export const EditProfileModal=({isOpen,onClose})=>{
+    const initialRef = useRef()
+    const finalRef = useRef()
+    const {editUserData,setEditUserData,loggedInUser,setLoggedInUser}=usePost()
+
+    const editUser=async()=>{
+        console.log(editUserData,"edit user");
```
Comment (deepak29-git) [inline my-tweet/src/components/edit-profile-modal.js:23]: remove console after debugging

**Review by deepak29-git — APPROVED** [review]
code structure is good and variable name is also good

File: my-tweet/src/context/post-context.js (hunk 1 above)
Comment (tushargoyal812) [reply my-tweet/src/context/post-context.js:16]: ok

File: my-tweet/src/components/userProfile.js (hunk 2 above)
Comment (tushargoyal812) [reply my-tweet/src/components/userProfile.js:14]: done

File: my-tweet/src/components/edit-profile-modal.js (hunk 5 above)
Comment (tushargoyal812) [reply my-tweet/src/components/edit-profile-modal.js:23]: done

File: my-tweet/src/pages/following.js (hunk 3 above)
Comment (tushargoyal812) [reply my-tweet/src/pages/following.js:24]: ok

File: my-tweet/src/components/edit-profile-modal.js (hunk 4 above)
Comment (tushargoyal812) [reply my-tweet/src/components/edit-profile-modal.js:33]: done

**Pull request merged**